backend/infrahub/core/task/task.py

```diff
@@ -91,3 +108,83 @@ async def query(
             )
 
         return {"count": count, "edges": nodes}
+
+
+class NewTask:
+    @classmethod
+    async def query(
+        cls,
+        fields: dict[str, Any],
+        related_nodes: list[str],
+        branch: str | None = None,
+        limit: int | None = None,
+        offset: int = 0,
+    ) -> dict[str, Any]:
+        nodes: list[dict] = []
+        count = None
+
+        log_fields = get_nested_dict(nested_dict=fields, keys=["edges", "node", "logs", "edges", "node"])
+        logs_flow: dict[str, list[PrefectLog]] = defaultdict(list)
+
+        async with get_client(sync_client=False) as client:
+            tags = [TAG_NAMESPACE]
+
+            if branch:
+                tags.append(WorkflowTag.BRANCH.render(identifier=branch))
+
+            # We only support one related node for now, need to investigate HOW (and IF) we can support more
+            if related_nodes:
+                tags.append(WorkflowTag.RELATED_NODE.render(identifier=related_nodes[0]))
+
+            flow_run_filters = FlowRunFilter(
+                tags=FlowRunFilterTags(all_=tags),
+            )
+
+            flows = await client.read_flow_runs(
+                flow_run_filter=flow_run_filters,
+                limit=limit,
+                offset=offset,
+                sort=FlowRunSort.START_TIME_DESC,
+            )
+
+            # For now count will just return the number of objects in the response
+            # it won't work well with pagination but it doesn't look like Prefect provide a good option to count all flows
```

Comment:
Perhaps it's not included in the client but possible:

http://localhost:4200/docs#/Flow%20Runs/count_flow_runs_flow_runs_count_post

I haven't tested this but see that there are two endpoint called count. I'd guess that one of them is what we'd want.

Reply:
interesting, I'll check it out